### PR TITLE
Refine SOF and hda machine driver to make SOF work with hda machine driver

### PR DESCRIPTION
**Pull request by RanderWang**
Hda Machine driver needs hda info from sof platform. Refine both components to share hda info

### CODE REVIEW

File: sound/soc/intel/boards/Kconfig [hunk 1]
```diff
@@ -326,6 +326,10 @@ config SND_SOC_INTEL_KBL_DA7219_MAX98357A_MACH
 	  create an alsa sound card for DA7219 + MAX98357A I2S audio codec.
 	  Say Y if you have such a device.
 
+endif ## SND_SOC_INTEL_SKYLAKE
+
+if SND_SOC_INTEL_SKYLAKE || SND_SOC_SOF_INTEL
+
```
Comment (keyonjie) [inline sound/soc/intel/boards/Kconfig:332]: better to use SND_SOC_SOF_HDA

File: sound/soc/sof/sof-pci-dev.c [hunk 2]
```diff
@@ -245,23 +245,11 @@ static int sof_pci_probe(struct pci_dev *pci,
...
-		dev_name(&sof_pdata->pdev_mach->dev));
+	sof_pdata->platform = "sof-audio";
 
 	/* register sof-audio platform driver */
 	ret = sof_create_platform_device(priv);
 	if (ret) {
-		platform_device_unregister(sof_pdata->pdev_mach);
 		dev_err(dev, "error: failed to create platform device!\n");
 		goto release_regions;
 	}
```
Comment (keyonjie) [inline sound/soc/sof/sof-pci-dev.c:255]: please add machine device register change for sof-spi-dev.c also.

Comment (RanderWang) [reply sound/soc/sof/sof-pci-dev.c:255]: get it

File: sound/soc/intel/boards/Kconfig (hunk 1 above)
Comment (RanderWang) [reply sound/soc/intel/boards/Kconfig:332]: get it

**Comment by keyonjie** [comment]
@RanderWang during trying this on my leafhill, I fixed a issue here also, please merge it to you series:

``` c

diff --git a/sound/soc/sof/sof-pci-dev.c b/sound/soc/sof/sof-pci-dev.c
index df873f3b8d50..e01ddb17f488 100644
--- a/sound/soc/sof/sof-pci-dev.c
+++ b/sound/soc/sof/sof-pci-dev.c
@@ -229,9 +229,10 @@ static int sof_pci_probe(struct pci_dev *pci,
                if (ret < 0)
                        goto release_regions;
 #else
-               dev_warn(dev, "No matching ASoC machine driver found - using HDA codec\n");
+               dev_warn(dev, "No matching ASoC machine driver found - "
+                        "fall back to use HDA codec\n");
                mach = snd_soc_acpi_intel_hda_machines;
-               mach->sof_fw_filename = desc->machines->sof_fw_filename;
+               mach->sof_fw_filename = desc->nocodec_fw_filename;
 #endif
        }
 #endif
```

**Comment by xiulipan** [comment]
@keyonjie 

> ++ dev_warn(dev, "No matching ASoC machine driver found - "
> ++ "fall back to use HDA codec\n");

80 char is rule is exception for any debug strings. They should be kept in oneline for user to grep them.

> However, never break user-visible strings such as printk messages, because that breaks the ability to grep for them. (https://www.kernel.org/doc/html/latest/process/coding-style.html#breaking-long-lines-and-strings)

**Comment by keyonjie** [comment]
> @keyonjie
> 
> > ++ dev_warn(dev, "No matching ASoC machine driver found - "
> > ++ "fall back to use HDA codec\n");
> 
> 80 char is rule is exception for any debug strings. They should be kept in oneline for user to grep them.

good point. @RanderWang please use something like:
--               dev_warn(dev, "No matching ASoC machine driver found - using HDA codec\n");
++               dev_warn(dev, "No matched ASoC machine driver found - falling back to HDA codec\n");

**Comment by RanderWang** [comment]
update my patch, thanks for review

**Comment by keyonjie** [comment]
LGTM, thanks.

**Comment by lgirdwood** [comment]
@RanderWang @keyonjie since we are modifying probe, I'd like to see a single point for PCI and ACPI probe between Intel SOF drivers and Intel Legacy drivers (to remove the need for blacklist).  We then select either legacy or SOF based on DMI name, defaulting to legacy. Lets do this after your national holidays.

**Pull request merged**